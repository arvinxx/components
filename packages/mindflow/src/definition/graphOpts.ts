import type { Options } from '@antv/x6/lib/graph/options';
import { minimapOpts } from './minimapOpts';
import { port } from './port';
import './edge';

/**
 * 生成图的配置项
 * @param container
 * @param minimapCtn
 */
export const graphOpts = (container, minimapCtn): Partial<Options.Manual> => ({
  container,
  background: {
    color: '#fafafa',
  },
  interacting: {
    // nodeMovable: false,
    edgeLabelMovable: false,
    edgeMovable: false,
  },
  autoResize: true,
  mousewheel: true,
  scroller: {
    enabled: true,
    pannable: true,
    autoResize: true,
  },
  minimap: minimapOpts(minimapCtn),
  // @ts-ignore
  onPortRendered: port,
});
