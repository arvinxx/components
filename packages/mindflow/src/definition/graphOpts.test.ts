import { graphOpts } from './graphOpts';

describe('graphOpts', () => {
  it('返回正常值', () => {
    const ctn = document.createElement('div');
    const minimap = document.createElement('div');
    expect(graphOpts(ctn, minimap)).toMatchSnapshot();
  });
});
