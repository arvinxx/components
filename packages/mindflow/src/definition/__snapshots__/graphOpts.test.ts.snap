// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphOpts 返回正常值 1`] = `
Object {
  "autoResize": true,
  "background": Object {
    "color": "#fafafa",
  },
  "container": <div />,
  "grid": 1,
  "interacting": Object {
    "edgeLabelMovable": false,
    "edgeMovable": false,
  },
  "minimap": Object {
    "container": <div />,
    "enabled": true,
    "graphOptions": Object {
      "async": true,
      "createCellView": [Function],
      "getCellView": [Function],
    },
    "height": 100,
    "maxScale": 1,
    "scalable": false,
    "width": 120,
  },
  "mousewheel": true,
  "onEdgeLabelRendered": [Function],
  "onPortRendered": [Function],
  "scroller": Object {
    "autoResize": true,
    "enabled": true,
    "pannable": true,
  },
}
`;
